app/policies/licenses/user_policy.rb

```diff
@@ -24,15 +44,40 @@ def show?
       end
     end
 
-    def update?
-      verify_permissions!('license.user.update')
-      verify_environment!
+    def attach?
+      verify_permissions!('license.users.attach')
+      verify_environment!(
+        # NOTE(ezekg) This seems weird, but we want to allow attaching shared users
+        #             to global licenses, but not vice-versa.
+        strict: license.environment.nil?,
+      )
+
+      case bearer
+      in role: Role(:admin | :developer | :sales_agent | :support_agent | :read_only | :environment)
+        allow!
+      in role: Role(:product) if license.product == bearer
+        allow!
+      in role: Role(:user) if license.owner == bearer
+        allow!
```

Comment:
We should only allow this when the policy or license is unprotected. This is a privilege escalation vector, where a license owner knows the ID of another user, but cannot read anything about them. Yet, if they attach that user to one of their licenses, they can now read the user because the user is now associated to them. Guessing IDs is out of the question, so this isn't a huge vector, but I still want this behavior to be explicit via an unprotected policy or license.

Reply:
Better yet, we could default user permissions to not allow attaching/detaching users to licenses. I prefer this approach, as it makes it even more explicit. We should still assert that the license is unprotected, in addition, since we do that everywhere else the license is mutated.